pennylane/devices/qubit/sampling.py

```diff
@@ -547,36 +547,22 @@ def _sample_state_jax(
     Returns:
         ndarray[int]: Sample values of the shape (shots, num_wires)
     """
-    # pylint: disable=import-outside-toplevel
-    import jax
-    import jax.numpy as jnp
-
     if prng_key is None:
         prng_key = jax.random.PRNGKey(np.random.default_rng(seed).integers(100000))
 
-    total_indices = len(state.shape) - is_state_batched
-    state_wires = qml.wires.Wires(range(total_indices))
-
-    wires_to_sample = wires or state_wires
-    num_wires = len(wires_to_sample)
-    basis_states = np.arange(2**num_wires)
-
-    flat_state = flatten_state(state, total_indices)
-    with qml.queuing.QueuingManager.stop_recording():
-        probs = qml.probs(wires=wires_to_sample).process_state(flat_state, state_wires)
+    basis_states = jnp.arange(2**num_wires)
 
     if is_state_batched:
-        keys = jax_random_split(prng_key, num=len(state))
+        keys = jax.random.split(prng_key, num=probs.shape[0])
         samples = jnp.array(
             [
                 jax.random.choice(_key, basis_states, shape=(shots,), p=prob)
                 for _key, prob in zip(keys, probs)
             ]
         )
     else:
-        _, key = jax_random_split(prng_key)
-        samples = jax.random.choice(key, basis_states, shape=(shots,), p=probs)
+        samples = jax.random.choice(prng_key, basis_states, shape=(shots,), p=probs)
```

Comment:
We should continue splitting the PRNGKey here, because otherwise the samples between each execution will not differ, which we need for MCMs to work correctly.